tardis/configuration/utilities.py

```diff
@@ -6,12 +6,12 @@ def yaml_load_decorator(cls):
         def class_factory(loader, node):
             new_cls = cls
             if isinstance(node, yaml.nodes.MappingNode):
-                parameters = loader.construct_mapping(node)
+                parameters = loader.construct_mapping(node, deep=True)
                 new_cls = cls(**parameters)
             elif isinstance(node, yaml.nodes.ScalarNode):
                 new_cls = cls()
             elif isinstance(node, yaml.nodes.SequenceNode):
-                parameters = loader.construct_sequence(node)
+                parameters = loader.construct_sequence(node, deep=True)
```

Comment:
This is necessary since MFA is using nested lists and dictionaries, when initialised using a yaml tag. This function is only used in the unittest. In real life it will be evaluated by COBalD, which already supports eager evaluation of yaml tags.